File: Basic Programming/Input-Output/C++/Maximum borders.cpp
/*
             ...........................................................
               Solver       :   Arif Khan
               Problem      :   Maximum borders
               Level        :   Easy
               Judge        :   HackerEarth
               Date         :   01/12/2023
               Time         :   0.02651 sec
               Memory       :   2 KB
               Difficulty   :   Easy
               problem_link :   https://www.hackerearth.com/practice/basic-programming/input-output/basics-of-input-output/practice-problems/algorithm/maximum-border-9767e14c/
               github       :   https://github.com/arifkhan1990
               portfolio    :   https://arif-portfolio-orcin.vercel.app/
               gamil        :   arifkhanshubro@gmail.com
               phone        :   01960518551
             ...........................................................
*/

#include<bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int t;
    cin >> t;

    while(t--) {
        int n , m;
        cin >> n >> m;

        vector<string> mat(n);
        for (int i = 0; i < n; ++i) {
            cin >> mat[i];
        }

        int ans = 0;

        for (const string& line : mat){
            int cur = 0;
            for (char x : line){
                cur = (x == '#') ? cur + 1: 0;
                ans = max(ans, cur);
            }
        }

        for (int col = 0; col < m; col++) {
            int cur = 0;
            for (int row = 0; row < n; ++row) {
                cur = (mat[row][col] == '#') ? cur + 1 : 0;
                ans = max(ans, cur);
            }
        }

        cout << ans << endl;
    }
    return 0;
}

File: Basic Programming/Input-Output/Python/Maximum borders.py
"""
             ...........................................................
               Solver       :   Arif Khan
               Problem      :   Maximum borders
               Level        :   Easy
               Judge        :   HackerEarth
               Date         :   01/12/2023
               Time         :   1.35144 sec
               Memory       :   6888 KB
               Difficulty   :   Easy
               problem_link :   https://www.hackerearth.com/practice/basic-programming/input-output/basics-of-input-output/practice-problems/algorithm/maximum-border-9767e14c/
               github       :   https://github.com/arifkhan1990
               portfolio    :   https://arif-portfolio-orcin.vercel.app/
               gamil        :   arifkhanshubro@gmail.com
               phone        :   01960518551
             ...........................................................
"""

t = int(input())

for _ in range(t):
    # Number of rows and columns
    n, m = map(int, input().split())

    matrix = []
    # Input matrix
    for _ in range(n):
        row = list(input().strip())
        matrix.append(row)
    
    ans = 0
    for i in range(n):
        curr_mx = 0
        for j in range(m):
            if matrix[i][j] == "#":
                curr_mx += 1
            else:
                ans = max(ans, curr_mx)
                curr_mx = 0

    for i in range(m):
        curr_mx = 0
        for j in range(n):
            if matrix[j][i] == "#":
                curr_mx += 1
            else:
                ans = max(ans, curr_mx)
                curr_mx = 0
    print(ans)